src/lsdb/loaders/dataframe/dataframe_catalog_loader.py

```diff
@@ -174,42 +186,22 @@ def _generate_dask_df_and_map(
         # Mapping HEALPix pixels to the respective Dataframe indices
         ddf_pixel_map: Dict[HealpixPixel, int] = {}
 
-        for hp_pixel_index, hp_pixel_info in enumerate(pixel_map.items()):
-            hp_pixel, (_, pixels) = hp_pixel_info
+        for hp_pixel_index, hp_pixel in enumerate(pixel_list):
             # Store HEALPix pixel in map
             ddf_pixel_map[hp_pixel] = hp_pixel_index
-            # Obtain Dataframe for current HEALPix pixel
-            pixel_dfs.append(self._get_dataframe_for_healpix(hp_pixel, pixels))
+            # Obtain Dataframe for current HEALPix pixel, using NESTED characteristics.
+            left_bound = healpix_to_hipscat_id(hp_pixel.order, hp_pixel.pixel)
+            right_bound = healpix_to_hipscat_id(hp_pixel.order, hp_pixel.pixel + 1)
+            pixel_df = self.dataframe.loc[
+                (self.dataframe.index >= left_bound) & (self.dataframe.index < right_bound)
+            ]
+            pixel_dfs.append(_append_partition_information_to_dataframe(pixel_df, hp_pixel))
 
         # Generate Dask Dataframe with the original schema and desired backend
         pixel_list = list(ddf_pixel_map.keys())
```

Comment:
Can we delete this assignment and use the `pixel_list` provided to the function?